### PR TITLE
Test lazy distance calculation in distance and closeness features.

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp [hunk 1]
```diff
@@ -24,16 +66,47 @@ DistanceClosenessFixture::DistanceClosenessFixture(size_t fooCnt, size_t barCnt,
         SimpleTermData term;
         term.setUniqueId(fooCnt + i + 1);
         term.addField(fieldId).setHandle(barHandles.back());
+        if (!query_tensor.empty()) {
+            term.set_query_tensor_name("qbar");
+        }
         queryEnv.getTerms().push_back(term);
     }
+    if (!query_tensor.empty()) {
+        attr_tensor = create_attribute_tensor("bar", "tensor(x[2])", docid_limit);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp:75]: Consider renaming variable to `tensor_attr`

File: searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.h [hunk 2]
```diff
@@ -55,7 +55,16 @@ struct DistanceClosenessFixture : BlueprintFactoryFixture, IndexEnvironmentFixtu
     RankProgram::UP          rankProgram;
     std::vector<TermFieldHandle> fooHandles;
     std::vector<TermFieldHandle> barHandles;
-    DistanceClosenessFixture(size_t fooCnt, size_t barCnt, const Labels &labels, const vespalib::string &featureName);
+    std::shared_ptr<search::tensor::DenseTensorAttribute> attr_tensor;
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.h:58]: Consider renaming member variable to `tensor_attr`.

**Review by toregge — APPROVED** [review]

File: searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp [hunk 3]
```diff
@@ -1,14 +1,56 @@
...
+using vespalib::eval::TensorSpec;
+using vespalib::eval::Value;
+using vespalib::eval::ValueType;
 
 namespace search::features::test {
 
+namespace {
+
+std::shared_ptr<DenseTensorAttribute>
+create_attribute_tensor(const vespalib::string& attr_name,
```
[the remark this answers is not in the document]
Comment (geirst) [reply searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp]: Done

File: searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp (hunk 1 above)
Comment (geirst) [reply searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.cpp]: Done

File: searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.h (hunk 2 above)
Comment (geirst) [reply searchlib/src/vespa/searchlib/test/features/distance_closeness_fixture.h]: Done

**Review by toregge — APPROVED** [review]

**Comment by geirst** [comment]
This adds unit tests for #22016.

**Pull request merged**